packages/pluggableWidgets/accordion-web/src/components/AccordionGroup.tsx

```diff
@@ -0,0 +1,50 @@
+import { createElement, ReactElement, ReactNode, useCallback, useContext, useEffect, useRef, useState } from "react";
+import { AccordionGroupsDispatch } from "./Accordion";
+
+export interface AccGroup {
+    header: ReactNode;
+    content: ReactNode;
+    collapsed?: boolean;
+    visible: boolean;
+}
+
+export interface AccordionGroupProps {
+    group: AccGroup;
+}
+
+export default function AccordionGroup(props: AccordionGroupProps): ReactElement | null {
+    const { group } = props;
+
+    const dispatch = useContext(AccordionGroupsDispatch);
+    const previousVisiblePropValue = useRef(group.visible);
+
+    const [divContentMounted, setDivContentMounted] = useState(group.visible && !group.collapsed);
+
+    useEffect(() => {
+        if (group.visible !== previousVisiblePropValue.current || !group.collapsed) {
+            previousVisiblePropValue.current = group.visible;
+            setDivContentMounted(group.visible && !group.collapsed);
+        }
+    }, [group, setDivContentMounted]);
+
+    const toggleContentVisibility = useCallback(() => {
+        if (group.collapsed) {
+            dispatch?.({ type: "expand", group });
+        } else {
+            dispatch?.({ type: "collapse", group });
+        }
+    }, [group, dispatch]);
+
+    if (!group.visible) {
+        return null;
+    }
+
+    return (
+        <section>
+            <header onClick={dispatch ? toggleContentVisibility : undefined}>{group.header}</header>
+            <div style={{ display: dispatch && group.collapsed ? "none" : undefined }}>
```

Comment:
Is there a reason this is done through inline styling render than conditional rendering on the div through JavaScript? Animations?

Reply:
I will move this to functional styles in the next story. :)